omod-1.11/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_11/LivingPatientSearchHandler1_11Test.java

```diff
@@ -0,0 +1,137 @@
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("q", "");
+		SimpleObject result = deserialize(handle(req));
+		List<Object> patients =  result.get("results");
+		Assert.assertEquals(0, patients.size());
+	} 
+	
+	@Test
+	public void shouldNotReturnPatientsIfQParamIsNotSet() throws Exception {	
+		Assert.assertEquals(8, patientService.getAllPatients().size());
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("includeDead", "True");
+		
+		SimpleObject result = deserialize(handle(req));
+		List<Object> patients =  result.get("results");
+		Assert.assertEquals(0, patients.size());
+	} 
+	
+	@Test
+	public void shouldReturnBothDeadAndLivingPatientsIfIncludeDeadIsSetToTrue() throws Exception {
+		patientService.getPatientByUuid(RestTestConstants1_11.PATIENT_UUID).setDead(true);
+		
+		Assert.assertEquals(8, patientService.getAllPatients().size());
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("q", "Moz");
+		req.addParameter("includeDead", "True");	
+		
+		SimpleObject result = deserialize(handle(req));
+		List<Object> patients =  result.get("results");
+		Assert.assertEquals(3, patients.size());
+	} 
+	
+	@Test
+	public void shouldReturnOnlyLivingPatientsBydefaultIfIncludeDeadIsNotSet() throws Exception {
+		patientService.getPatientByUuid(RestTestConstants1_11.PATIENT_UUID).setDead(true);
+		
+		Assert.assertEquals(8, patientService.getAllPatients().size());
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("q", "Moz");
+				
+		SimpleObject result = deserialize(handle(req));
+		List<Object> patients =  result.get("results");
+		Assert.assertEquals(2, patients.size());
+				
+	}
+	
+	@Test
+	public void shouldReturnOnlyLivingPatientsOnWrongIncludeDeadParamValue() throws Exception {
+		patientService.getPatientByUuid(RestTestConstants1_11.PATIENT_UUID).setDead(true);
+		
+		Assert.assertEquals(8, patientService.getAllPatients().size());
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("q", "Moz");
+		req.addParameter("includeDead", "wrongx");
+		
+		SimpleObject result = deserialize(handle(req));
+		List<Object> patients =  result.get("results");
+		Assert.assertEquals(2, patients.size());	
+	}
+	
+	@Test
+	public void shouldReturnOnlyLivingPatientsIfIncludeDeadParamIsSetToFalse() throws Exception {
+		patientService.getPatientByUuid(RestTestConstants1_11.PATIENT_UUID).setDead(true);
+		
+		Assert.assertEquals(8, patientService.getAllPatients().size());
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		req.addParameter("q", "Moz");
+		req.addParameter("includeDead", "False");
+		
+		SimpleObject result = deserialize(handle(req));
+		List<Object> patients =  result.get("results");
+		Assert.assertEquals(2, patients.size());	
+	}
+	
+	
+	@Test
+	public void shouldNotReturnPatientsIfNoPatientMatchesQParam() throws Exception {
```

Comment:
Are your tests about the q or includeDead parameter?

Reply:
they are about the include dead, but since i also made use of the Q param , i thought it was necessary i also do a test on the Q param. do you mean to say this test is unnecessary?

Reply:
hi @dkayiwa . more clarification on this comments.. did you intend to mean this test is unnecesarry??. since in my changes i have both the Q and IncludeDead param , i would think its necesarry do also do some test on the Q param